ios/Classes/SwiftManageCalendarEventsPlugin.swift

```diff
@@ -165,8 +165,11 @@ public class SwiftManageCalendarEventsPlugin: NSObject, FlutterPlugin {
     }
 
     private func requestPermissions() {
-        eventStore.requestAccess(to: .event, completion: {
+        eventStore.requestAccess(to: .event, completion: { [weak self]
             (accessGranted: Bool, error: Error?) in
+            if accessGranted, let weakSelf = self {
+                weakSelf.eventStore =  EKEventStore()
+            }
```

Comment:
I think this is the most common way to weakify the self:

```suggestion
            if accessGranted, let self = self {
                self.eventStore =  EKEventStore()
            }
```